WalletWasabi.Gui/CommandLine/DropWalletCacheCommand.cs

```diff
@@ -0,0 +1,114 @@
+using NBitcoin;
+using Mono.Options;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Services;
+using WalletWasabi.KeyManagement;
+using WalletWasabi.Gui;
+
+namespace WalletWasabi.Gui.CommandLine
+{
+	internal class DropWalletCacheCommand : Command
+	{
+		public string WalletName { get; set; }
+		public Network Network { get; set; }
+		public Daemon Daemon { get; }
+		public bool ShowHelp { get; set; }
+
+		public DropWalletCacheCommand(Daemon daemon)
+			: base("dropwalletcache", "Drops the wallet cache for the specified wallet.")
+		{
+			Daemon = daemon;
+			Network = Network.Main;
+
+			Options = new OptionSet()
+			{
+				"usage: dropwalletcache --wallet:WalletName",
+				"",
+				"Drops the wallet cache for the specified wallet.",
+				"eg: ./wassabee dropwalletcache --wallet:MyWalletName --network:main",
+				"",
+				{ "w|wallet=", "The name of the wallet file.", x =>  WalletName = x },
+				{ "n|network=", "The network for the given file (main, test, reg). Default=mainnet.", x => Network = GetNetwork(x) },
+				{ "h|help", "Show Help.", v => ShowHelp = true },
+			};
+		}
+
+		public override async Task<int> InvokeAsync(IEnumerable<string> args)
+		{
+			var error = false;
+			try
+			{
+				var extra = Options.Parse(args);
+				if (ShowHelp)
+				{
+					Options.WriteOptionDescriptions(CommandSet.Out);
+				}
+				else if (string.IsNullOrWhiteSpace(WalletName))
+				{
+					Logging.Logger.LogCritical<DropWalletCacheCommand>("Missing required argument `--wallet=WalletName`.");
+					Logging.Logger.LogCritical<DropWalletCacheCommand>("Use `dropwalletcache --help` for details.");
+					error = true;
+				}
+				else if (Network is null)
+				{
+					Logging.Logger.LogCritical<DropWalletCacheCommand>("Invalid argument `--network=Network`.");
+					Logging.Logger.LogCritical<DropWalletCacheCommand>("Use `dropwalletcache --help` for details.");
+					error = true;
+				}
+				else if (!string.IsNullOrWhiteSpace(WalletName))
+				{
+					KeyManager km = Daemon.TryGetKeyManagerFromWalletName(WalletName);
+					if (km is null)
+					{
+						error = true;
+					}
+					else
+					{
+						await Daemon.Global.InitializeNoWalletAsync();
```

Comment:
I am not sure about this, I would have prefer to let it as it was before because this does a lot of unnecessary things like starting Tor, the HWI process, the synchronizer, and the connection with peers. @nopara73, are you sure this is what you want?

Reply:
You are right, this smells, too. In this case, I don't know what's the clean solution is.

Reply:
We only really need to load the `BitcoinStore`, so we could copy the lines from the initalize function and do 

```c#
var BitcoinStore = new BitcoinStore();
await BitcoinStore.InitializeAsync(Path.Combine(Daemon.Global.DataDir, "BitcoinStore"), Network);
```

However, if we ever update the location or how this is done we would need to remember to change it in both places. We could make it it's own function though.

Reply:
Maybe elevate out the bitcoinstore initialization of the global, where nowallet initialization uses that and this uses that, too?